File: android_app/app/src/main/java/com/filescanner/app/util/LibraryLogic.kt
package com.filescanner.app.util

/**
 * 列表/合集分页相关的纯计算，从 LibraryViewModel 中提取，便于 JUnit 单测，且与 Android 运行时无关。
 * 行为与原内联实现完全一致，仅做位置重构。
 */
object LibraryLogic {

    /**
     * 计算总页数：至少 1 页（即使总数为 0，也返回 1，使 UI 可显示“第 1 页 / 共 1 页”）。
     * 等价于 ((total + pageSize - 1) / pageSize).coerceAtLeast(1)。
     */
    fun computePageCount(total: Int, pageSize: Int): Int {
        require(pageSize > 0) { "pageSize 必须 > 0，实际 $pageSize" }
        return ((total + pageSize - 1) / pageSize).coerceAtLeast(1)
    }

    /**
     * 解析合集“排除书名”输入：按逗号/换行切分，去空白、丢弃空项。
     * 等价于 raw.split(Regex("[,\n]+")).map { it.trim() }.filter { it.isNotEmpty() }。
     */
    fun parseExcludeNames(raw: String): List<String> {
        return raw.split(Regex("[,\n]+")).map { it.trim() }.filter { it.isNotEmpty() }
    }

    /**
     * 当前页码越界回退：若 page 超过最后一页（pageCount-1），夹到最后一页；小于 0 夹到 0。
     * 等价于 page.coerceIn(0, (pageCount - 1).coerceAtLeast(0))。
     */
    fun adjustPage(page: Int, pageCount: Int): Int {
        return page.coerceIn(0, (pageCount - 1).coerceAtLeast(0))
    }
}

File: ios_app/FileScanner/Core/LibraryLogic.swift
import Foundation

/// 列表/合集分页相关的纯计算（对齐 Android `LibraryLogic`）。
enum LibraryLogic {
    /// 计算总页数：至少 1 页。
    static func computePageCount(total: Int, pageSize: Int) -> Int {
        precondition(pageSize > 0, "pageSize 必须 > 0")
        return ((total + pageSize - 1) / pageSize).coerceAtLeast(1)
    }

    /// 解析合集「排除书名」输入：按逗号/换行切分，去空白、丢弃空项。
    static func parseExcludeNames(_ raw: String) -> [String] {
        raw.components(separatedBy: CharacterSet(charactersIn: ",\n")).map { $0.trimmingCharacters(in: .whitespaces) }.filter { !$0.isEmpty }
    }

    /// 当前页码越界回退：夹到 [0, pageCount-1]。
    static func adjustPage(_ page: Int, pageCount: Int) -> Int {
        return page.clamped(to: 0...(pageCount - 1).coerceAtLeast(0))
    }
}
